src/views/home/index.tsx

```diff
@@ -31,6 +36,34 @@ export const HomeView = () => {
     };
   }, [marketEmitter, midPriceInUSD, tokenMap]);
 
+  // TODO: 1. is this the right place for this and 2. is it okay to use two useEffects?
+  useEffect(() => {
+    async function populateOAuthToken() {
+      const code = gh.parseOAuthCode() || null;
+
+      if (code) {
+        const oauthToken = await gh.getOAuthToken(code);
+        setOAuthToken(oauthToken);
+      }      
+    }
+
+    if (!oauthToken) {
+      populateOAuthToken();
+    }
+  }, [oauthToken, setOAuthToken])

```

Comment:
im trying to understand, you're getting oauthToken from inside the populateOAuthToken function,
may i know the main logic here?

Reply:
This was a struggle. From what I read I needed to call the function from within the `useEffect` method. But this is probably something I'm butchering. Lmk if there's a better way to approach this. Otherwise maybe we tradeoff for the hack and I'll make sure to follow up with the research.

Reply:
i think what its doing is if `oauthToken` is not yet set, populate it and set it, so seems right to me, what doesnt is declaring it again as const in line 45

Reply:
Is my latest change to this snippet related? Maybe I'm oversimplifying.

```ts
const [oauthToken, setOAuthToken] = useState<string>("");

// ...

useEffect(() => {
    async function populateOAuthToken() {
      const code = gh.parseOAuthCode() || null;

      if (code) {
        const token = await gh.getOAuthToken({code: code}) || "";
        setOAuthToken(token);
      }      
    }

    if (oauthToken === "") {
      populateOAuthToken();
    }
  }, [oauthToken, setOAuthToken])
  ```

Reply:
ah ok thats better